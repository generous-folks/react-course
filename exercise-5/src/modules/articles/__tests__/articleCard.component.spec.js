import React from 'react';

import { shallow } from 'enzyme';

import { ArticleCard } from '../articleCard.component';

import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

let wrapper;

const article = {
  id: 'abc',
  slug: 'foo',
  name: 'bar',
  year: '1',
  image: 'baz',
};

const getWrapper = () => shallow(<ArticleCard article={article} />);

beforeEach(() => {
  jest.clearAllMocks();
  wrapper = getWrapper();
});

describe('<ArticleCard />', () => {
  describe('Snapshot', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Markup checks', () => {
    it('should contain the correct markup', () => {
      expect(wrapper.find(Grid).exists()).toBeTruthy();
      expect(wrapper.find(Card).exists()).toBeTruthy();
      expect(wrapper.find(CardMedia).prop('image')).toBe(article.image);
      expect(
        wrapper
          .find(CardContent)
          .find(Typography)
          .first()
          .text(),
      ).toBe(article.name);
      expect(
        wrapper
          .find(CardContent)
          .find(Typography)
          .last()
          .text(),
      ).toBe(article.year);
      expect(
        wrapper
          .find(CardActions)
          .find(Button)
          .first()
          .text(),
      ).toBe('Lorem');
      expect(
        wrapper
          .find(CardActions)
          .find(Button)
          .last()
          .text(),
      ).toBe('Ipsum');
    });
  });
});
