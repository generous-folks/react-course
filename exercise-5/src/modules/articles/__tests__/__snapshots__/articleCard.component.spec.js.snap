// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> Snapshot should render correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  md={4}
  sm={6}
  xs={12}
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-card-1"
  >
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-cardMedia-2"
      image="baz"
      title="bar"
    />
    <WithStyles(ForwardRef(CardContent))
      className="makeStyles-cardContent-3"
    >
      <WithStyles(ForwardRef(Typography))
        component="h2"
        gutterBottom={true}
        variant="h5"
      >
        bar
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-cardDescription-4"
      >
        <WithStyles(ForwardRef(Typography))>
          1
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))>
          1
           $
        </WithStyles(ForwardRef(Typography))>
      </div>
    </WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(CardActions))>
      <WithStyles(ForwardRef(Button))
        color="secondary"
        onClick={[Function]}
        size="small"
        variant="outlined"
      >
        Lorem
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        size="small"
        variant="outlined"
      >
        Ipsum
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(CardActions))>
  </WithStyles(ForwardRef(Card))>
</WithStyles(ForwardRef(Grid))>
`;
